include/envoy/request_id_extension/request_id_extension.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+#include "envoy/http/header_map.h"
+
+namespace Envoy {
+namespace RequestIDExtension {
+
+enum class TraceStatus { NoTrace, Sampled, Client, Forced };
+
+/**
+ * Abstract request id utilities for getting/setting the request IDs and tracing status of requests
+ */
+class Utilities {
+public:
+  virtual ~Utilities() = default;
+
+  /**
+   * Directly set a request ID into the provided request headers. Override any previous request ID
+   * if any.
+   * @param request_headers supplies the incoming request headers for setting a request ID.
+   */
+  virtual void setRequestID(Http::RequestHeaderMap& request_headers) PURE;
+
+  /**
+   * Ensure that a request is configured with a request id.
```

Comment:
I'm still not sure what the difference is between set and ensure. Can you clarify? I thought you had discussed removing this also?